corpdir-srv-jpa/src/main/java/com/github/bordertech/corpdir/jpa/util/MapperUtil.java

```diff
@@ -204,7 +208,7 @@ public static String cleanApiKey(final String key) {
 	 * @param <T> the entity
 	 * @return the entity
 	 */
-	public static <T extends PersistKeyIdObject> T getEntityByBusinessKey(final EntityManager em, final String businessKey, final Class<T> clazz) {
+	public static <T extends PersistIdObject> T getEntityByBusinessKey(final EntityManager em, final String businessKey, final Class<T> clazz) {
```

Comment:
If getting an `Entity` via `BusinessKey` it should always be a `PersistKeyId`. Is there a reason this needs to be a `PersistIdObject`?

Reply:
Yes, it is to retrieve an `ApiIdObject` object by `BusinessKey`. I need it so as to create/update models like `UnitType`

Reply:
Sorry...I know we have spoken about this already.

Why does the `MapperUtil` method called in `getEntity` need to be changed to `byKeyId` instead of `byApiId`? The `ID` being passed into that method should always be an `ID` (not a `businessKey`).

Calling this other method does cause the confusion of the `byKeyId` and `ByBusinessKey` methods  now returning `ApiIdObject` instead of `ObjectKeyIdObject`.